ts_forecasting_pipeline/speccing.py

```diff
@@ -173,19 +193,24 @@ def load_series(self) -> pd.Series:
             )
 
         # Keep the most recent observation
-        series_orig = (
+        series = (
             series_orig.sort_values(by=["horizon"], ascending=True)
             .drop_duplicates(subset=["datetime"], keep="first")
             .sort_values(by=["datetime"])
         )
-        series_orig.set_index("datetime", drop=True, inplace=True)
+        series.set_index("datetime", drop=True, inplace=True)
 
-        if series_orig.index.tzinfo is None:
+        if series.index.tzinfo is None:
             if self.original_tz is not None:
-                series_orig.index = series_orig.index.tz_localize(self.original_tz)
+                series.index = series_orig.index.tz_localize(self.original_tz)
         else:
-            series_orig.index = series_orig.index.tz_convert(self.original_tz)
-        series = series_orig.resample("15T").mean()
+            series.index = series.index.tz_convert(self.original_tz)
+
+        if expected_frequency is not None:
+            series = series_orig.resample(
+                timedelta_to_pandas_freq_str(expected_frequency)
+            ).mean()
```

Comment:
You consistently take the mean of the Series values when resampling, but that may not always be the appropriate resampling policy for every feature. As we discussed, it could be to take the sum, to take a nansum, a nanmean, to pad/ffill or to assume zeros, etc. Perhaps we should leave it as a function for the caller to decide what policy to apply.

Reply:
Rolled into a new ticket.